### DISCUSSION TITLE
Public Code

### DISCUSSION THREAD

**Discussion by ghandylan** [community/community#19134] [Copilot Conversations] [answered]
![unknown-3](https://user-images.githubusercontent.com/80078209/174955938-ce2fcec0-335e-4b63-a38f-d93a7d349fa2.png)

Does selecting "Block" on Suggestions matching public code affect Copilot's performance? Will Copilot provide less/dumber code if I block this feature?

**Comment by melvincwng** [comment, marked as answer]
@ghandylan, from my understanding in the [Github Copilot FAQ](https://github.com/features/copilot/), selecting "Block" essentially prevents GitHub Copilot from suggesting code that can be found publicly.

Anyway, I wouldn't worry too much about it as the settings can be changed even after being set. Please refer to [this link](https://docs.github.com/en/copilot/configuring-github-copilot/configuring-github-copilot-in-visual-studio-code) for more info.

**Reply by melvincwng** [reply]
@ghandylan, hope this helps.

Kindly 'Mark as answer' if this issue is resolved/addressed.

Have a good day ahead 😄

**Comment by ecker00** [comment]
I'm curious why this option exists, what problem does this option solve?
